src/Controller/DashboardController.php

```diff
@@ -0,0 +1,230 @@
+   * time of retrieve the post and add new post.
+   *
+   *   @var object
+   */
+  public $postService;
+
+  /**
+   * This constructor initializes object of DashboardController Class also
+   * provides access to EntityManagerInterface object .
+   *
+   *   @param  object $em
+   *     It is to manage persistance and retriveal Entity object from Database.
+   *
+   *   @return void
+   *     Constructor returns nothing .
+   */
+  public function __construct(EntityManagerInterface $em)
+  {
+    $this->em = $em;
+    $this->user = new User();
+    $this->post = new PostData();
+    $this->postService = new Posts();
+    $this->userRepo = $this->em->getRepository(User::class);
+    $this->postRepo = $this->em->getRepository(PostData::class);
+    if (isset($_COOKIE["userId"])) {
+      $this->userId = $_COOKIE["userId"];
+    }
+  }
+
+  /**
+   * This method is to show the posts of all users, the lists of other
+   * users and the information of user,who logged in currently.Basically it
+   * takes the user to the dashboard page after login succesfully.
+   *
+   * @Route("/dashboard", name = "dashboard")
+   *  It takes the user to the dashboard page after login succesfully and it
+   *  redirects to login page if user try to reach this page without login.
+   *
+   *   @param object $rq
+   *     This Request object is to handles the user credentials.
+   *
+   *   @return Response
+   *     Based on the validation the response is to either dashboard page or
+   *     the login page with the proper error messages.
+   */
+  public function index(): Response
+  {
+    if (isset($_COOKIE["active"])) {
+      return $this->render("dashboard.html.twig",[
+        "allpost" => $this->postRepo->findAll(),
+        "userInfo" => $this->userRepo->findOneBy(["userId" => $this->userId]),
+        "otherUserInfo" => $this->userRepo->findAllByUserIdExcept($this->userId),
+      ]);
+    }
+    return $this->redirect("/");
+  }
+
+  /**
+   * This method used when user try add a new post, it stores the post data
+   * in database and redirects to dashboard with loading the new post.
+   *
+   * @Route("/makepost", name = "makepost")
+   *  This route does not take user to a new page it redirects user to the
+   *  dashboard after store the new post data.
+   *
+   *   @param object $rq
+   *     This Request object is to handles the post data.
+   *
+   *   @return Response
+   *     Redirects to the dashboard page.
+   */
+  public function makePost(Request $rq): Response
+  {
+    echo $this->userId;
+    if ($rq != NULL ) {
+      $this->post->setPostId($this->postService->generatePostId($this->userId));
+      $this->post->setPostContent($rq->request->get("postContent"));
+      $image = $rq->files->get("postImage");
+      if ($image != NULL) {
+        $this->post->setPostImage($this->postService->postImgStoring($image));
+      }
+      $this->user = $this->em->getRepository(User::class)->findOneBy([
```

Comment:
You have declared it in constructor. So why not directly use $this->userRepo?

Reply:
Changes Done.